File: CMOWebApi.Models/ComparetiveModel/PreviousGovernmentEntryModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CMOWebApi.Models.ComparetiveModel
{
	public class PreviousGovernmentEntryModel
	{
		public int Id { get; set; }
		public Nullable<int> Code { get; set; }
		public Nullable<int> DepartmentCode { get; set; }
		public Nullable<long> ParameterCode { get; set; }
		public Nullable<int> KPICategoryCode { get; set; }
		public bool IsActive { get; set; }
		public bool IsDeleted { get; set; }
		public Nullable<System.DateTime> CreatedDate { get; set; }
		public Nullable<int> CreatedBy { get; set; }
		public Nullable<System.DateTime> ModifiedDate { get; set; }
		public Nullable<int> ModifiedBy { get; set; }
		public List<PreviousGovernmentEntryParameterMappingModel> PreviousGovernmentEntryParameterMappingModel { get; set; }

	}

	public class PreviousGovernmentEntryParameterMappingModel
	{
		public long Id { get; set; }
		public Nullable<int> PreviousGovtId { get; set; }
		public Nullable<int> YearCode { get; set; }
		public string YearName { get; set; }
		public string PhysicalValue { get; set; }
		public string FinancialValue { get; set; }
	}

	public partial class PreviousGovernmentEntryListModel
	{
		public int Id { get; set; }
		public Nullable<int> Code { get; set; }
		public Nullable<int> DepartmentCode { get; set; }
		public Nullable<int> KPICategoryCode { get; set; }
		public bool IsActive { get; set; }
		public bool IsDeleted { get; set; }
		public string DepartmentTitle { get; set; }
		public string KPICategoryName { get; set; }
		public Nullable<System.DateTime> CreatedDate { get; set; }
		public Nullable<System.DateTime> ModifiedDate { get; set; }
		public string modifiedbyName { get; set; }
		public string PhysicalParameter { get; set; }
	}
}

File: ANGULAR/ANGULAR/src/app/Shared/Model/Camparetive/previous-government-entry-model.ts
export class PreviousGovernmentEntryModel {
		Id: number;
		Code: number | null;
		DepartmentCode: number | string | null;
		ParameterCode: number | string | null;
		KPICategoryCode: number | string | null;
		IsActive: boolean;
		IsDeleted: boolean;
		CreatedDate: Date | string | null;
		CreatedBy: number | null;
		ModifiedDate: Date | string | null;
		ModifiedBy: number | null;
		PreviousGovernmentEntryParameterMappingModel: PreviousGovernmentEntryParameterMappingModel[] = [];

	}

	export class PreviousGovernmentEntryParameterMappingModel {
		Id: number;
		PreviousGovtId: number | null;
    YearCode: number | string | null;
    YearName: string;
		PhysicalValue: string;
		FinancialValue: string;
    constructor(){
      this.PreviousGovtId=0;
      this.Id=0;
    }
	}
export class PreviousGovernmentEntryListModel {
  Id: number;
  Code: number | null;
  DepartmentCode: number | null;
  KPICategoryCode: number | null;
  IsActive: boolean;
  IsDeleted: boolean;
  DepartmentTitle: string;
  KPICategoryName: string;
  CreatedDate: Date | string | null;
  ModifiedDate: Date | string | null;
  modifiedbyName: string;
  PhysicalParameter: string;
}


